rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -292,10 +292,14 @@ class LabelBatch : public MetaDataBatch
     }
     std::shared_ptr<MetaDataBatch> clone(bool copy_contents) override
     {
-        if(copy_contents)
+        if(copy_contents) {
```

Comment:
I think both branches are making copies of the labelsBatch. Why do we need to treat them separate?

Reply:
The if condition - Creates copy of the complete metadata batch along with labels, bbox and all the info
The else condition - Creates a new instance of the metadata batch, it copies only the info and does not copy the labels, bbox and mask as it is expected to be filled in by the metanodes

Reply:
I am not sure if this is the right practice. When is the clone used?

Reply:
Clone is used in the output routine
1. We fill the metadata batch struct with metadata of respective images using look up
2. We need to create a copy of the metadata batch to be stored into the ring buffer
3. To create the copy we use clone.

If metanodes are enabled - We are creating a new instance of the metadata batch, it copies only the info and does not copy the labels, bbox and mask as it is expected to be filled in by the metanodes
else - We need to clone the entire metadata along with labels, bbox and all the info